node/src/neighborhood/mod.rs

```diff
@@ -4640,17 +4642,17 @@ mod tests {
 
         System::current().stop();
         system.run();
-        let min_hops_count_params = min_hops_count_params_arc.lock().unwrap();
-        assert_eq!(*min_hops_count_params, vec![()]);
+        let min_hops_params = min_hops_params_arc.lock().unwrap();
+        assert_eq!(*min_hops_params, vec![()]);
```

Comment:
We said you want to use the assertion message to test the state of the Actor after the validation. It'd be strong enough assertion. Your version does not prove things at all. It just says that somewhere within the code somebody called `min_hops()`, however, what was done with the value or what the action influenced is a big unknown. 

Only if you prove that the state of the actor, with the concern of the min_hops parameter, was the same before and after we can speak of some conclusiveness.    

Bonus: it might also mean that in the end you won't really need your param assertion tool set. If I were in your shoes, I'd probably get rid of those rather unnecessary pieces of code with the unit type as the asserting medium.

Reply:
Done 👍